### PR TITLE
GDB-10042 Ensure the VMSS will always be able to resolve addresses in the Private DNS Zone

### PR DESCRIPTION
**Pull request by viktor-ribchev**
## Description

Ensure the VMSS will always be able to resolve addresses in the Private DNS Zone
This handles cases where the VNet has custom DNS servers set. 

## Related Issues

GDB-10042

## Changes

Added the MS internal DNS to be the default DNS for the VMSS.

## Screenshots (if applicable)

image

## Checklist

- [X] I have tested these changes thoroughly.
- [X] My code follows the project's coding style.
- [X] I have added appropriate comments to my code, especially in complex areas.
- [X] All new and existing tests passed locally.

### CODE REVIEW

File: modules/graphdb/main.tf [hunk 1]
```diff
@@ -82,6 +82,9 @@ resource "azurerm_linux_virtual_machine_scale_set" "graphdb" {
   network_interface {
     name    = "nic-${var.resource_name_prefix}-vmss"
     primary = true
+    # Ensures the VMSS will always resolve addresses in the Private DNS Zone.
+    # https://learn.microsoft.com/en-us/azure/virtual-network/what-is-ip-address-168-63-129-16
+    dns_servers = ["168.63.129.16"]
```
[the remark this answers is not in the document]
Comment (viktor-ribchev) [reply modules/graphdb/main.tf:87]: Weird works without too

[the remark this answers is not in the document]
Comment (viktor-ribchev) [reply modules/graphdb/main.tf]: made it configurable

File: modules/graphdb/main.tf [hunk 2]
```diff
@@ -2,6 +2,12 @@
 # Linux VMs scale set for GraphDB
 #
 
+# Ensures the VMSS will always resolve addresses in the Private DNS Zone.
+# https://learn.microsoft.com/en-us/azure/virtual-network/what-is-ip-address-168-63-129-16
+locals {
+  dns_servers = contains(var.vmss_dns_servers, "168.63.129.16") ? var.vmss_dns_servers : concat(var.vmss_dns_servers, ["168.63.129.16"])
```
Comment (mihailradkov) [inline modules/graphdb/main.tf:8]: Checkout Terraform's `setunion`, i.e.

```hcl
dns_servers = setunion(var.vmss_dns_servers, ["168.63.129.16"])
```

Comment (viktor-ribchev) [reply modules/graphdb/main.tf]: cool didn't know about that.

**Review by mihailradkov — APPROVED** [review]

**Pull request merged**